Rewrite `Gameplay/Multiplayer` (#6813)
ppy/osu-wiki: wiki/Gameplay/Multiplayer/en.md

The three buttons above the [chat console](/wiki/Client/Interface/Chat_console) can be used in the main lobby:

| Title | Description |
| :-: | :-- |
| `Back to Menu` | Leave the lobby and return to the main menu. |
| `New Game` | Create a new game. See below for more details. |
| `Quick Join` | Join an available match *randomly* based on the player's current pp rank range. |

## Creating a new game

![](img/Multi_new_game.jpg "Setting up configuration for a new game")

| Title | Description |
| :-: | :-- |
| `Game Name` | The name of the match. The default value is `{account name}'s game`. |
| `Require password to join` | Make the match private. |
| `Password` | Set a password for the match. Only appears if the `Require password to join` button was enabled. |
| `Max Players` | Total number of players allowed to join the match. The default is 8 players, with a range from 2 to a total of 16 players, including the player as host. It can be adjusted later by locking/unlocking the slots. |

Pressing the `1. Start Game` button will create the match with the beatmap playing in the jukebox as the match's beatmap. The button `2. Cancel` brings the player back to the lobby.

> ## Match setup

Reviewer: this section mixes neutral and informal writing styles with the `here's what you can see` additions

| **Green (ready)** | The player is ready. Game modifiers can't be changed in this state. A player can press the `Not Ready` button to return to the white state. The host will have a `Start Game` button after pressing ready. |
| **Light Blue (playing)** | The player is playing. A `[playing]` notice will be appended at the end of player's name until the match ends. |

### Match settings

![](img/multi-match-settings.jpg "Match settings")

This section is about the match settings. The host can change the game name, the beatmap, the team mode and other stuff.

#### Change game name and password

Here, the host can change the name for the match. Additionally, the password for the match can be adjusted via the `Change Password` button.

#### Beatmap

![](img/Multi_host_song_change.jpg "When the host is changing songs, this bar will become a temporary placeholder.")

The beatmap section shows the beatmap to be played for this match.

When choosing the beatmap, the host will enter a menu similar to the song selection, but they will not be able to apply any mods.

> ![](img/Multi_SS.jpg "Example of how the song selection will look like for the host.")

Reviewer: ```suggestion
![](img/Multi_SS.jpg "Example of what the song selection will look like for the host.")
```

![](img/Multi_beatmapinfo.jpg "Beatmap selection when hovered over.")

On hovering over the beatmap, there will be an extra dialogue displaying further information about the beatmap. These are as follows:

| Value | Description |
| :-: | :-- |
| `BPM` | Beats per minute |
| `Length` | Length of the beatmap |
| `Last Played` | When the beatmap was played the last time |
| `CS` | Circle Size |
| `AR` | Approach Rate |
| `OD` | Overall Difficulty |
| `HP` | HP Drain Rate |
| `Stars` | Star Difficulty |

If the player does not have the beatmap, one of these statuses will be shown:

| Beatmap status | Description |
| :-: | :-- |
| `Ranked` / `Approved` / `Pending` / `Graveyard` | The [beatmap category](/wiki/Beatmap/Category). Clicking on the beatmap card will open the beatmap page on the website, where it can be [downloaded](/wiki/Beatmap#downloading-beatmaps). |
| `Not uploaded or not up-to-date` | The beatmap is not available for download. The host should find the link (in the beatmap listing) for the player if the issue persists. |

> | `Cannot update the beatmap` | The [host is probably playing a modified version](https://osu.ppy.sh/community/forums/topics/135726 "peppy's reply on a bug report"). |

Reviewer: you can remove the link safely, along with `probably`
```suggestion
| `Cannot update the beatmap` | The host is playing a modified version of the beatmap. |
```

#### Mods

![](img/Multi_mods_host.jpg "Host mod options")

![](img/Multi_mods_player.jpg "Player (osu!) mods options (with Free Mods enabled)")

The lower left section is the mods section. This section will show the [game mods](/wiki/Game_modifier) used for this match.

The host can enable `Free Mods` to allow players to freely choose any mod combination, except ones that change gameplay speed ([Double Time (DT)](/wiki/Game_modifier/Double_Time), [Nightcore (NC)](/wiki/Game_modifier/Nightcore) and [Half Time (HT)](/wiki/Game_modifier/Half_Time)).

#### Team mode {#team-mode-match-setup}

> *For more information about team modes, see [Team mode](#team-mode-while-playing).*

Reviewer 2: i'm not entirely sure what to do with this tbh, because i can't think of any alternative titles for these headings

#### Multiplayer leaderboard

The leaderboard to the side of the screen shows different live statistics for each player based on the match settings.

For `Head-to-head` and `Team VS` matches:

| Win condition | Statistics displayed |
| :-- | :-- |
| `Score` / `Score v2` | Individual score, combo multiplier, judgements |
| `Accuracy` | Accuracy, individual score, combo multiplier, judgements |
| `Combo` | Current combo counts |

For `Tag co-op` and `Tag-team VS`:

| `Score` / `Score v2` | Team score, individual score, judgements |
| `Accuracy` | Team average accuracy, individual score, judgements |
| `Combo` | Current combo count, individual score, judgements |

If a player fails during a `Tag co-op` match, the match ends and the players return to the match lobby. In `Tag-team VS`, the opposing team receives a win by default instead. Failed players do not contribute towards the team unless they have been revived by the end of the match.

If a player quits when playing in these modes, the game will perfectly play their part for them, but not contribute anything to the team. If all players of a team quit, the opposing team will win by default.

> Players who have failed or quit will be placed at the bottom of the leaderboard with their scores sorted separately from other alive players.

Reviewer: I'd pick a wording other than "alive players", perhaps just "from other players" will suffice

##### Result

![](img/Multi_grade.jpg "Head to Head results screen")

At the end of the match, the winner's avatar is shown in a big winner's box, along with the match leaderboard to the right

The [grade](/wiki/Gameplay/Grade) for the play is placed behind the result like a watermark. Other players' results can be checked by clicking on their panel on the leaderboard.

Each player will receive their match result as a private message in the `#userlog` tab in [chat](/wiki/Client/Interface/Chat_console).

#### Team Vs

##### Interface

![](img/Multi_team_vs.jpg "Team Vs interface")

`Team Vs` pits two teams, Blue and Red, against each other. Fulfill the win condition as closely as possible to reach the top spot in the team and compete with the other team.

Blue is on the left and Red is on the right. A crown at the centre moves left and right to show which team is currently winning.

Changing teams can be done while in the lobby. There is no handicap for teams with fewer members, and it is entirely possible to start a match with only one team.

##### Result

![](img/Multi_team_grade.jpg "Team Vs results screen")

The winning team is announced at the top of the results screen.

> Only players who are alive by the end of the match count towards their teams' totals.

Reviewer 2: i've tentatively reworded this from "counting to team score", but i'm not sure if that's the case for the accuracy and combo win conditions as well

`Tag Coop` makes all players in the match work together to pass the beatmap, one combo at a time. `Tag Team Vs` is just `Tag Coop` but with two teams playing against each other.

If a player quits during the match, [Auto](/wiki/Game_modifier/Auto) will take over for the player that left.
In this mode, the players' slot placement in the lobby determines the order they play in.

For Tag Coop, all players will share the same health bar.
If the health bar was depleted completely (failed state), the match ends and all players return to the Match Setup immediately without result.
A green arrow will show the player who is currently playing, and their name will be shown at the lower right. Warning arrows indicate when it is about to be the player's turn and the hit objects will be coloured. The player can set a fixed combo colour at the match setup the [Tag Colour](#tag-colour) section in the match lobby.

For Tag Team Vs, each team has its own respective health bar that are visible to teammates only.
If the health bar was depleted completely (failed state), the match ends by giving victory to the other team.
All players must spin the spinner, or the combo will break.

> If [Easy (EZ)](/wiki/Game_modifier/Easy) game modifier was used, the two health bar refill provided by the game modifier will still occur but will be depleted completely immediately after *just* to fail the beatmap.

Reviewer 2: i kinda have no idea what this was supposed to mean, but i didn't think it was very important so i've removed it for now

Reviewer: uhhhhhhh me neither, but I think it means that a player won't have any benefits from the HP bar restoration (= will fail instantly anyway)

The result screen only appear if the beatmap was cleared successfully.
For `Tag Coop`, the result shown is the total score, the [judgements](/wiki/Gameplay/Judgement) and accuracy of the team. Individual results are not shown, and can only be seen in the match history. The result screen only appears if the beatmap was cleared successfully, unlike other multiplayer modes.

For Tag Team Vs result screen, refer to Team Vs's result screen explanation.
For `Tag Team Vs` results screen, refer to `Tag Team Vs`'s [result screen explanation](#team-vs).

## Trivia

### History

![](img/Multi_Mania_unpatched.jpg "Screenshot of Lobby of an unpatched osu! when osu!mania was released to public \(08 October 2012, 2012-10-08\)")
::: Infobox
![](img/Multi_Mania_unpatched.jpg "Screenshot of the main lobby when osu!mania was first released \(October 8, 2012\), but not yet playable in multiplayer")
:::

- Match Setup used to only allows up to a total of 8 players only.
- osu!mania game mode was noted as `3` in Lobby for players that have not yet updated their osu! to the latest release patch (which enables osu!mania mode with no ranking support yet).

> - No Video was the *only* game modifier allowed for the players to use; however, it was removed in favour of its placement in the Visual Settings.

Reviewer 2: does this imply that this setting only existed in the visual settings panel before also being added to the regular settings?

Reviewer 2: https://github.com/ppy/osu-wiki/issues/7165

# Multiplayer

**Multiplayer** (sometimes shortened to *Multi*) is a mode in which up to 16 players can compete against each other or as a team on a single map decided by the host.

The [osu!academy](/wiki/Community/Video_series/osu!academy) covered this topic in [Episode 6](https://www.youtube.com/watch?v=QPTLyG7O8ak) along with the [online users panel](/wiki/Client/Interface/Chat_console#extended-chat-console).

## How to play

*Notice: A player can't play in multiplayer mode if his account is [silenced](/wiki/Silence).*

In order to join other players, the player must be signed in to [Bancho](/wiki/Bancho_(server)) via his osu! account. From the main menu, the player needs to do the following steps to get into the main lobby:

1. Click the `Play` button or press `P`.
2. Click the `Multi` button or press `M`.

*Note: If the player loses connection to Bancho while playing in multiplayer mode, he will be directed back to the main menu.*

## Main lobby

*Notice: An active [osu!supporter](/wiki/osu!supporter) tag is required to enter the lobby while using the `Cutting Edge` build.*

> 

Reviewer: yes

> 

Reviewer 2: could add this to the supporter article

| Title | Description |
| :-: | :-- |
| `All`/`osu!`/`osu!taiko`/`osu!catch`/ `osu!mania` | Show only matches either with the currently selected [game mode](/wiki/Game_mode) or from all game modes. |
| `Owned Beatmaps` | Show only matches if the player owns the currently played [beatmap](/wiki/Beatmap). |
| `Show Full` | Show matches with full slots. Do note that these matches can't be entered as long as there are no available slots. |
| `Search` | Click on the search bar and search specifically for the used beatmap or the host's name. When searching, all filters will be ignored temporarily except `Show In-progress`. |
| `Games with Friends` | Only show matches that the player's friends are in. This filter option will overwrite all other filter options when enabled. |
| `Show Locked` | Show matches that are locked. Do note that to enter the match, a proper password must be given. |
| `Show In-progress` | Show matches that are currently played. If the match contains empty slots, it is possible to enter it. The match text colour will change to grey with `(In progress)` appended to the back of the title until the match ends. |

### Matches

> In the middle, there are rows displaying the available matches. We will cover the functions from left to right.

Author: This `We` should be changed to a more neutral form. I will do this later.

| :-: | :-- |
| `Game Name` | The name of the match. The default value is `{account name}'s game`. |
| `Require password to join` | Make the match private. |
| `Password` | Set a password for the match. Only appears if the `Require password to join` button is enabled. |
| `Max Players` | The total number of players (including the host) allowed to join the match. A range from 2 to 16 can be selected, with a default of 8. This can be adjusted later by locking/unlocking player slots. |

Pressing the `1. Start Game` button will create the match with the currently playing song as the beatmap. The `2. Cancel` button brings the player back to the lobby.

## Match setup

::: Infobox
![](img/Multi_roomhost.jpg "The match lobby viewed as a host")
:::

::: Infobox
![](img/Multi_roomplayer.jpg "The match lobby viewed as a player")
:::

After going into or creating a match, the match setup will be displayed. The following sections will focus on the different parts of the screen starting from top to bottom and going left to right.

While there is no direct button for it, the [options menu](/wiki/Client/Options) can still be accessed by pressing `Ctrl` + `O` while in a match lobby.

> Note that the the match setup screen looks slightly different for hosts and players (see the two images to the right).

Author: ```suggestion
Note that the match setup screen looks slightly different for hosts and players (see the two images to the right).
```
